integration/spark/build.gradle

```diff
@@ -71,19 +74,25 @@ dependencies {
     compileOnly "org.projectlombok:lombok:${lombokVersion}"
     annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
 
-    implementation "io.openlineage:openlineage-java:${project.version}"
+    implementation("io.openlineage:openlineage-java:${project.version}"){
+        exclude group: 'com.fasterxml.jackson.core'
+        exclude group: 'com.fasterxml.jackson.datatype'
+    }
     implementation 'org.javassist:javassist:3.27.0-GA'
-    implementation 'com.github.ok2c.hc5:hc5-async-json:0.2.1'
     implementation 'org.apache.httpcomponents.client5:httpclient5:5.0.3'
-    implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.module:jackson-module-scala_2.11:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${jacksonDatatypeVersion}"
+    compileOnly "com.fasterxml.jackson.core:jackson-databind:${jacksonDatabindVersion}"
+    compileOnly "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
+    compileOnly "com.fasterxml.jackson.module:jackson-module-scala_2.11:${jacksonModuleScalaVersion}"
```

Comment:
`LogicalPlanSerializer` wouldn't  work with `compileOnly jackson-module-scala_2.11`  for 2.4.7 and for 3.1. Due to `relocate` in `shadowJar`. In compile time we have openlineage prefix, and if we bound it to our jar, it would work fine for 2.4.7. If we run it for spark3, we would get   
```
java.lang.NoSuchMethodError: scala.Product.$init$(Lscala/Product;)V
      at io.openlineage.spark.agent.lifecycle.plan.LogicalRDDVisitorTest.testApply(LogicalRDDVisitorTest.java:49)
```
Cause it loads scala 2.11 in 2.12 env